### PR TITLE
Dox++

### PR DESCRIPTION
**Pull request by nyalldawson**
Add some missing documentation for some ambiguous API.

### CODE REVIEW

File: python/core/auto_generated/qgsrendercontext.sip.in [hunk 1]
```diff
@@ -111,6 +111,11 @@ Returns the destination QPainter for the render operation.
 %Docstring
 Returns the current coordinate transform for the context, or an invalid
 transform is no coordinate transformation is required.
+
+This usually represents the coordinate transform required to transform a layer
```
Comment (m-kuhn) [inline python/core/auto_generated/qgsrendercontext.sip.in]: I find "usually" a tricky term in an API. I.e. can I trust this behavior and if not, under what conditions not / what do I need to do.

File: python/core/auto_generated/qgsrendercontext.sip.in [hunk 2]
```diff
@@ -163,6 +168,22 @@ during rendering operations, e.g. for resolving relative symbol paths.
 %End
 
     const QgsRectangle &extent() const;
+%Docstring
+When rendering a map layer, calling this method returns the "clipping"
+extent for the layer (in the layer's CRS).
+
+This extent is a "worst-case" scenario, which is guaranteed to cover the complete
+visible portion of the layer when it is rendered to a map. It is often larger
+then the actual visible portion of that layer.
```
Comment (m-kuhn) [inline python/core/auto_generated/qgsrendercontext.sip.in]: larger ghan

**Comment by m-kuhn** [comment]
Thanks for taking care of api docs :)

File: src/core/qgsrendercontext.h [hunk 3]
```diff
@@ -269,9 +288,39 @@ class CORE_EXPORT QgsRenderContext
...
+
     void setMapToPixel( const QgsMapToPixel &mtp ) {mMapToPixel = mtp;}
+
+    /**
+     * Sets the map \a extent to be rendered.
+     *
+     * When rendering a map layer, this extent must be in the corresponding layer's CRS.
+     *
+     * \see extent()
+     */
```
Comment (wonder-sk) [inline src/core/qgsrendercontext.h]: Is this comment also related to setExtent() ? maybe it should be merged with the comment block below as sipify does not seem to use this one...
